include/wil/resource.h

```diff
@@ -597,7 +597,7 @@ namespace wil
     template <typename T>
     struct smart_pointer_details<Microsoft::WRL::ComPtr<T>>
     {
-        typedef T* pointer;
+        using pointer = T*;

```

Comment:
There's a `typedef` in the non-specialized definition right above this one. We should change either both or neither.